core/src/test/java/org/apache/iceberg/rest/RESTCatalogAdapter.java

```diff
@@ -320,7 +321,10 @@ public <T extends RESTResponse> T handleRequest(
 
       case DROP_TABLE:
         {
-          CatalogHandlers.dropTable(catalog, identFromPathVars(vars));
+          CatalogHandlers.dropTable(
+              catalog,
+              identFromPathVars(vars),
+              PropertyUtil.propertyAsBoolean(vars, "purgeRequested", false));
```

Comment:
we're passing `false` here if `purgeRequested` wasn't provided, but previously data was always purged due to `Catalog.dropTable(ident)` always using `purge=false`

Reply:
You mean always using `purge=true`?

Reply:
I think this is correct for the REST API. If `purgeRequested` was not set then it was not requested. This should be consistent behavior, since purge would cause a failure before.